### PR TITLE
Fix package entry point and publish type declarations

### LINKED ISSUE #58: Issues when including `music21j` in my project
Hello, I'm trying to use `music21j` in my project.

I'm a bit new to the node environment, but I figured out that using `webpack` was convenient for my purpose, so I created my project and then I run:

    npm install --save music21j

Then in one of my source files I tried:

    import { note } from "music21j"

But when I build the project using `npx webpack` I get various resolve errors like the following:

```
ERROR in ./node_modules/music21j/src/music21_modules.js
Module not found: Error: Can't resolve './music21/articulations' in '/home/matteo/git/nuvola/node_modules/music21j/src'
 @ ./node_modules/music21j/src/music21_modules.js 48:0-57 91:0-141:2
 @ ./src/core/melody/melody.js
 @ ./src/main.js
```

So I resorted to use the already built version provided (which however seems a bit of a hack to me, the above example should just work), by using

    import { note } from "music21j/releases/music21.debug"

I suspect that the issue might be related to the fact that the modules are written in typescript, but typescript doesn't get added to my dependencies nor get configured in the proper way. I don't know which is the proper way to handle this thing.

---

The second problem I'm facing is related with MIDI soundfonts. If I just build like in the above example I don't get any playback, and in the browser console I see the following error:

    Could not load soundfont; path was ../soundfonts/midi-js-soundfonts-master/FluidR3_GM//acoustic_grand_piano-ogg.js

The hack I found to make it work locally is to copy the files from `./node_modules/music21j/soundfonts` to `./dist/soundfonts` (I automated it using `copy-webpack-plugin`), but it doesn't work when I deploy to GitHub Pages (because of the level up `..` at the beginning of `../soundf....` and because I'm deploying a project page, so my base url is `mttbernardini.github.io/myproject`).

Is there a proper way to change the path of the soundfonts? I didn't find anything in the documentation and it seems to be hardcoded somewhere.

---

Let me know if you need more details, I'm willing to help.

Thank you in advance.

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
Thanks for your patience @mttbernardini, I've added you as a co-author here so that you're credited as a repo contributor.

Fixes #58 (I'll open an issue for the second issue you mentioned)

We were waiting to fix the entry point issue until we verified we could still consume music21j's declared types in other projects when using this slimmer import of `import * as music21 from "music21j";`, which is FANTASTIC. To that end, I had to adjust the entry-point based on some changes in #93 (`main.js` instead of `debug.js`)

***

@mscuthbert this is intended to supersede both #60 and the `refactor_for_load` branch. Whatever existing procedure you have to copy files from `build` to `releases` at release-time should pick up the type declarations.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -6,6 +6,9 @@
         "experimentalDecorators": true,
         "moduleResolution": "node",
         "sourceMap": true,
+        "declaration": true,
+        "declarationMap": true,
+        "declarationDir": "build",
```
Comment (mscuthbert) [inline tsconfig.json:11]: is this a problem since build isn't committed or will the map also be moved to releases?

File: package.json [hunk 2]
```diff
@@ -2,7 +2,8 @@
   "name": "music21j",
   "version": "0.11.11",
   "description": "A toolkit for computer-aided musicology, Javascript version",
-  "main": "releases/main.js",
+  "main": "releases/music21.debug.js",
+  "typings": "releases/main.d.ts",
```
Comment (jacobtylerwalls) [inline package.json]: Incredibly, I missed that `main.js` was not a file. And that we do need to declare the location of the typings.

**Comment by jacobtylerwalls** [comment]
There's also an infelicity that the typing declarations are duplicated under `build/src` in addition to `build/`, which I believe is because of the test suite transpilation. I experimented with using two different `tsconfig.json` files, one in src/ and one in tests/, but I didn't get very far.

**Comment by mscuthbert** [comment]
Very important PR.  Thanks @jacobtylerwalls !

**Pull request merged**